sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -0,0 +1,369 @@
+    # This number of tables doesnt slow down performance by much (15% compared to 500 tables)
+    # but allows the queue to be stable.
+    TABLES_CHUNK_SIZE = 50
+
+    def __init__(self, check, schemas_collection_interval):
+        self._check = check
+        self._log = check.log
+        self.schemas_per_db = {}
+
+        base_event = {
+            "host": None,
+            "agent_version": datadog_agent.get_version(),
+            "dbms": "sqlserver",
+            "kind": "sqlserver_databases",
+            "collection_interval": schemas_collection_interval,
+            "dbms_version": None,
+            "tags": self._check.non_internal_tags,
+            "cloud_metadata": self._check._config.cloud_metadata,
+        }
+        self._dataSubmitter = SubmitData(self._check.database_monitoring_metadata, base_event, self._log)
+
+    """Collects database information and schemas and submits to the agent's queue as dictionaries
+    schema dict
+    key/value:
+        "name": str
+        "id": str
+        "owner_name": str
+        "tables" : list of tables dicts
+            table
+            key/value:
+                "id" : str
+                "name" : str
+                columns: list of columns dicts
+                    columns
+                    key/value:
+                        "name": str
+                        "data_type": str
+                        "default": str
+                        "nullable": bool
+            indexes : list of index dicts
+                index
+                key/value:
+                    "name": str
+                    "type": str
+                    "is_unique": bool
+                    "is_primary_key": bool
+                    "is_unique_constraint": bool
+                    "is_disabled": bool,
+                    "column_names": str
+            foreign_keys : list of foreign key dicts
+                foreign_key
+                key/value:
+                    "foreign_key_name": str
+                    "referencing_table": str
+                    "referencing_column": str
+                    "referenced_table": str
+                    "referenced_column": str
+            partitions: partition dict
+                partition
+                key/value:
+                    "partition_count": int
+    """
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def collect_schemas_data(self):
+        self._dataSubmitter.reset()
+        self._dataSubmitter.set_base_event_data(
+            self._check.resolved_hostname,
+            self._check.non_internal_tags,
+            self._check._config.cloud_metadata,
+            "{},{}".format(
+                self._check.static_info_cache.get(STATIC_INFO_VERSION, ""),
+                self._check.static_info_cache.get(STATIC_INFO_ENGINE_EDITION, ""),
+            ),
+        )
+
+        # returns if to stop, True means stop iterating.
+        def fetch_schema_data(cursor, db_name):
```

Comment:
would add this tracking annotation here as well
```
@tracked_method(agent_check_getter=agent_check_getter)
```

Reply:
sure, will do